bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Cursor.java

```diff
@@ -300,7 +301,7 @@ public Cursor(Device device, ImageData source, int hotspotX, int hotspotY) {
 		if (hMask == 0) SWT.error(SWT.ERROR_NO_HANDLES);
 	} else {
 		ImageData mask = source.getTransparencyMask();
-		long [] result = Image.init(this.device, null, source, mask);
+		long [] result = Image.init(this.device, null, source, mask, DPIUtil.getDeviceZoom());
```

Comment:
In the beginning when we don't have any zoom info, we set it to this one.
But in this specific case, this zoom information is never utilized in the init method. because all this does is to set the handle for the zoom level in the map. but since the passed image is null here, and that block is never entered, it does nothing.

The most important thing is the imageData in this case. because the imageData is obtained using image.getImageData in the MenuSpyHandler and that gets the imageData at zoom level 100 (also in points basically). But overall the argument DPIUtil.getDeviceZoom() doesnt play any roll in this call here. We can also nullify it here.

Reply:
If it doesn't play any role and it can be nullified then please do it and add a small comment in the code explaining that the parameter will be ignored anyway.

Reply:
Hmm, I'll have to change the data type of the parameter to Integer in this case to make it nullable though. Is that okay?

Reply:
Done